docs/tutorials/course/fairbo.md

```diff
@@ -0,0 +1,506 @@
+# Fair Bayesian Optimization
+
+Given the increasing importance of machine learning (ML) in our lives, with a wide use of automated systems in several domains, such as financial lending, hiring, criminal justice, and college admissions, there has been a major concern for ML to unintentionally encode societal biases and result in systematic discrimination. For example, a classifier that is only tuned to maximize prediction accuracy may unfairly predict a high credit risk for some subgroups of the population applying for a loan. 
+
+In several real-world domains, accuracy is not the only objective of interest. The model should simultaneously give guarantees on other important aspects. Algorithmic fairness tries to find algorithms that not only keep a high level of accuracy but also enforce a certain level of fairness and avoid biases.
+
+The approach in this tutorial is based on the paper "Fair Bayesian Optimization". Link: https://arxiv.org/abs/2006.05109
+
+If you use this code please cite 'Fair Bayesian Optimization'. Valerio Perrone, Michele Donini, Muhammad Bilal Zafar, Robin Schmucker, Krishnaram Kenthapadi, Cédric Archambeau. AIES. 2021.
+
+In this tutorial we are going to use the German Credit Data (1). This dataset contains a binary classification task, where the goal is to predict if a person has "good" or "bad" credit risk. The dataset contains 20 attributes, including a sensitive features such as the (binary) gender of the person. In the following we download, load and explore this dataset.
+
+(1) Dua, D. and Graff, C. (2019). [UCI Machine Learning Repository](http://archive.ics.uci.edu/ml). Irvine, CA: University of California, School of Information and Computer Science.
+
+
+
+```python
+# AutoGluon and HPO tools
+import autogluon.core as ag
+import pandas as pd
+import numpy as np
+import random
+
+# Plots
+import matplotlib.pyplot as plt
+import seaborn as sns
+sns.set()
+
+# Fixing seed for reproducibility
+SEED = 123
+random.seed(SEED)
+np.random.seed(SEED)
+```
+
+## Dataset setup and visualization
+
+This code will automatically download the data from the UCI repository.
+You can download the data manually from here: [UCI german lending dataset link.](https://archive.ics.uci.edu/ml/datasets/statlog+(german+credit+data))
+
+
+```python
```

Comment:
You can see the CI did not evaluate any of the cells in your notebook from the logs here:  https://ci.gluon.ai/blue/rest/organizations/jenkins/pipelines/autogluon/branches/PR-1090/runs/2/nodes/153/steps/193/log/?start=0

This is because each cell that you actually want evaluated needs to be preceded by: 
"```{.python .input}"
instead of:  "```python"

Reply:
Your cells are currently simply being shown without being actually executed on the CI machine. 
However this may not be a bad thing. It seems this tutorial would take a while to execute during CI and would thus slow it down a lot?

This would not be good, so you could either: keep the cells as is (do not execute during CI, but this is unreliable failing to catch future bugs), or make this tutorial run more efficiently. If you choose the latter option, can you report how long it takes after you've updated it to run quicker?